### PR TITLE
fix catkin_lint issues

### PR DESCRIPTION
**Pull request by rhaschke**
Fix various catkin_lint issues and enable catkin_lint checker for Travis.

### CODE REVIEW

**Comment by jonbinney** [comment]
Wow, a ton of good fixes here!

**Comment by jonbinney** [comment]
I've tried to summarize the types of changes made in this PR, to convince myself its all fine (it is):
- Change package.xml to format 2
- Update run_depend to exec_depend
- Add REQUIRED for some find_package calls in cmake.
- Surpress warnings that raen't actuall a problem
- Add missing find_package calls in cmake
- Add missing catkin_depend's in cmake
- Only run coverage tests if CATKIN_ENABLE_TESTING is set
- Quote variable exapansions in cmake
- Add missing test_depends in package.xml
- Don't hardcode library install destination; use CATKIN_GLOBAL_LIB_DESTINATION
- "" For include install paths
- Install script that wasn't getting installed
- Include the right file (boost/bind)
- ompl_interface gets a system depend on OMPL in cmake catkin_package
- Re-order cmakelists to have find_package calls before include_directories calls
- Export plugin libraries via catkin_package in moveit_ros_control_interface
- Simplify build/run depends into one depend in package.xmls

File: moveit_ros/visualization/CMakeLists.txt [hunk 1]
```diff
@@ -98,8 +94,7 @@ include_directories(rviz_plugin_render_tools/include
 
 include_directories(SYSTEM
                     ${EIGEN3_INCLUDE_DIRS}
-                    ${QT_INCLUDE_DIR}
-                    ${OGRE_INCLUDE_DIRS})
```
[the remark this answers is not in the document]
Comment (rhaschke) [reply moveit_ros/visualization/CMakeLists.txt]: Exactly.

**Comment by rhaschke** [comment]
@jonbinney Thanks for the thorough review and nice summary!

**Pull request merged**